cmd/kubeadm/app/cmd/init.go

```diff
@@ -72,7 +71,7 @@ var (
 		You can now join any number of machines by running the following on each node
 		as root:
 
-		  kubeadm join --token {{.Token}} {{.MasterIP}}:{{.MasterPort}} --discovery-token-ca-cert-hash {{.CAPubKeyPin}}
+		  kubeadm join --token {{.Token}} {{.MasterEndpoint}} --discovery-token-ca-cert-hash {{.CAPubKeyPin}}
```

Comment:
I had to rely on parsing `kubeadm join` output in k-d-c at some point because `kubeadm` command line arguments were very much in flux back then and didn't restore the old behavior yet, but in this case this hack helped to catch the bug.